designs/lily/src/back.mjs

```diff
@@ -0,0 +1,567 @@
+          .move(points.floorOut)
+          .curve(points.floorOutCp2,points.kneeOutCp1,points.kneeOut)
+          .curve(points.kneeOutCp2, points.seatOut, waistOut)
+      else
+        return new Path()
+          .move(points.floorOut)
+          .curve(points.floorOutCp2,points.kneeOutCp1,points.kneeOut)
+          .curve(points.kneeOutCp2, points.seatOutCp1, points.seatOut)
+          .curve_(points.seatOutCp2, waistOut)
+  }  
+  /*
+   * Helper method to draw the outline path
+   */
+  const drawPath = () => {
+    let waistIn = points.styleWaistInLily || points.waistIn
+    return drawInseam()
+      .line(points.floorOut)
+      .join(drawOutseam())
+      .line(waistIn)
+      .line(points.crossSeamCurveStart)
+      .curve(points.crossSeamCurveCp1, points.crossSeamCurveCp2, points.fork)
+      .close()
+  }
+  /*
+   * Helper method to calculate the length of the cross seam
+   */
+  const crossSeamDelta = () =>
+    new Path()
+      .move(points.waistIn)
+      .line(points.crossSeamCurveStart)
+      .curve(points.crossSeamCurveCp1, points.crossSeamCurveCp2, points.fork)
+      .length() - measurements.crossSeamBack
+  /*
+   * Helper method to (re)draw the cross seam
+   */
+  const drawCrossSeam = () => {
+    points.crossSeamCurveStart = points.waistIn.shiftFractionTowards(
+      points.cbSeat,
+      options.crossSeamCurveStart
+    )
+    points.crossSeamCurveMax = utils.beamsIntersect(
+      points.waistIn,
+      points.cbSeat,
+      points.fork,
+      points.fork.shift(0, 1) // beams have infinite length anyway
+    )
+    points.crossSeamCurveCp1 = points.crossSeamCurveStart.shiftFractionTowards(
+      points.crossSeamCurveMax,
+      options.crossSeamCurveBend
+    )
+    points.crossSeamCurveCp2 = points.fork
+      .shiftFractionTowards(points.crossSeamCurveMax, options.crossSeamCurveBend)
+      .rotate(options.crossSeamCurveAngle, points.fork)
+  }
+
+  // majority of points re-used from titan
+
+/*   // Let's get to work
+  points.waistX = new Point(-1 * measurements.waistBackArc * (1 + options.waistEase), 0)
+  points.upperLegY = new Point(0, measurements.waistToUpperLeg)
+  points.seatX = new Point(-1 * measurements.seatBackArc * (1 + options.seatEase), 0)
+  points.seatY = new Point(0, measurements.waistToSeat)
+  points.seatOut = points.seatY
+  points.cbSeat = new Point(points.seatX.x, points.seatY.y)
+
+  // Determine fork location
+  points.fork = new Point(
+    measurements.seatBackArc * (1 + options.seatEase) * -1.25,
+    points.upperLegY.y * (1 + options.crotchDrop)
+  )
+
+  // Grainline location, map out center of knee and floor
+  points.grainlineTop = points.upperLegY.shiftFractionTowards(
+    points.fork,
+    options.grainlinePosition
+  )
+  points.knee = new Point(points.grainlineTop.x, measurements.waistToKnee)
+  points.floor = new Point(
+    points.grainlineTop.x,
+    measurements.waistToFloor * (1 + options.lengthBonus)
+  )
+  points.grainlineBottom = points.floor
```

Comment:
The grainline extends outside the pattern piece if lengthReduction is increased. You might want to set it later in the code after the lengthReduction adjustments and base it in relation to the reduced length.

Reply:
I've made a note, to be fixed in a later version.

Reply:
(I've reverted this back to a 'draft' pull request since the `flag` thing doesn't work yet.)